### ISSUE TITLE
Integrate Always Centred's functionality into this mod

### ISSUE THREAD

**Issue by p4535992** [coffiarts/FoundryVTT-hot-pan#8]
Always Centered is not follow anymore, the original author suggestion is to use "Monk's Common Display" 

_It is recommended that you consider use of [Monk's Common Display](https://github.com/ironmonk88/monks-common-display).
It is an excellent module and covers most of what Always Centred does._

**Comment by coffiarts** [comment]
Thanks for your suggestion. But this is not the first time that I am hearing this, and I do only _partially_ agree:

I agree in the following ways:

1. Monk's Common Display is a really great mod, especially for inplace/hybrid gaming. 
2. It is still under active development
3. Even Always Centred's author himself has recommend me to rather rely on MCD instead of on his own mod, which will probably be desupported from a certain time on in the future.

At the same time, however ...

1. Monk's mod is much more complex and does several things that not everybody needs, especially when there is _no shared screen_ involved at all!
2. Monk's mod in fact does _not_ do _exactly_ the same as Always Centred. I had a longer discussion with Always Centred's author about it. I've even encouraged him not to give up on his mod, and I have even contributed to it a bit in order to keep it alive.
3. My mod here (Hot Pan and Zoom!) declares AC only as an _optional_ dependency, so I see no harm in further recommending it.
5. My mod does not collide with either mod (at least not that I know of any conflict), so I see no conflict here.

In simpler terms: I've always loved AC for its simplicity! For non-hybid/non-inperson gaming, it does exactly the one thing it promises, and it does that in a very simple and elegant way.
That's why I wouldn't want to miss it.
Of course, people may find it a bit overdone to add it as a recommended mod to mine. I'll consider changing that in the future. Maybe I'll go even a step further an implement AC's functionality myself as an inbuilt option within Hot Pan & Zoom! But this is something for the future.

**Comment by p4535992** [comment]
If you agree i can try to put the always centered feature in this module directly (as a module settings to switch on/off). It would be one less module to have in my belly, and I could add controls for MDC

**Comment by coffiarts** [comment]
This sounds like a great idea (exactly what I was envisioning with my post above). If you feel encouraged to do that, and find the time for it, I'd be very happy.
However, as I assume that "putting the feature in this module" will probably mean to copy some code, I guess we should ask the author of [Always Centred](https://github.com/SDoehren/always-centred), [SDoehren](https://github.com/SDoehren/always-centred) for allowance.
If you don't mind, I will try to contact him.
I will also reopen this issue (and rename it a bit), so that we have a poper issue for a pull request.

**Comment by coffiarts** [comment]
Just asked SDoehren on Discord a sec ago. Waiting for a reply...

**Comment by SDoehren** [comment]
Hi Both,  

You are fully welcomed to lift any code from Always Centred.

The license is [Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License](http://creativecommons.org/licenses/by-nc-sa/4.0/), so I already give permission that way.   
Further to this, for this project I'm happy to lift the Attribution, NonCommercial, and ShareAlike requirements.  Thus making the code from Always Centred fully usable for free and without restriction within FoundryVTT-hot-pan.

I would encourage the use of a creative commons license in the spirit of the open community, but do not require it.

**Comment by coffiarts** [comment]
Hi @SDoehren 
Thank you very much for your permission. I appreciate very much that you are so open to our idea.
It hadn't come to my mind that you have already granted use of your code by your license, sorry. But nevertheless, it probably can't do harm to ask explicitely.
@p4535992 - As already stated, please feel free to provide a pull request (as I won't find time to do it myself right now). I will do my best to give you both proper credits for your contribution!

**Comment by coffiarts** [comment]
H there @p4535992 - Just wanted to ask if you're still planning on doing (or have already started with) this integration. Because otherwise, I see a good chance of starting it by myself within the next weeks.
Just wanted to make sure not to interfere with any activity on your side.
I'd be glad if you could give me a short bump. If I shouldn't hear from you, I assume that you haven't started yet and that I am free to go myself, ok?
I really like this idea of yours, and 'd love to provide a solution for it soon. Especially as Foundry v12 is picking up pace by now. This would be the perfect time to get that "mod fusion" done.
So thanks in advance four your reply!

**Comment by p4535992** [comment]
is on my to-do list , but low priority, I have old modules to fix and other inconveniences. I'm also a bit bogged down with v12 unfortunately, and I couldn't give you an estimate of when I could get my hands on it.

Free to take the “reins” in short.

**Comment by coffiarts** [comment]
Thanks for the update. I'll give it a try.

**Comment by coffiarts** [comment]
![image](https://github.com/coffiarts/FoundryVTT-hot-pan/assets/39616483/3e2a9bfc-520f-4888-a9d1-d911ef13c926)
It is almost complete. See the [dev branch](https://github.com/coffiarts/FoundryVTT-hot-pan/tree/dev) for details.
Only some cosmetics, finetuning and readme updates to be done.
I expect to release it within the next couple of days

**Comment by coffiarts** [comment]
Merge is done and released per today: https://github.com/coffiarts/FoundryVTT-hot-pan/releases/tag/latest

**Comment by p4535992** [comment]
> ![image](https://private-user-images.githubusercontent.com/39616483/338741298-3e2a9bfc-520f-4888-a9d1-d911ef13c926.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgyNjQwMDcsIm5iZiI6MTcxODI2MzcwNywicGF0aCI6Ii8zOTYxNjQ4My8zMzg3NDEyOTgtM2UyYTliZmMtNTIwZi00ODg4LWE5ZDEtZDkxMWVmMTNjOTI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEzVDA3MjgyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjMTA3MjVkOGE3YTQ2ZWM1Y2Q2NmJlZWRkYjlmZmZkMDQxNTMwMWFiNjg1MmQ2MjBlOTQwMzY2MDUzY2JhZTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SETmDu_XsuozxAirijz7AGMlUtjC8zWVTmJ2dUzQ-mA) It is almost complete. See the [dev branch](https://github.com/coffiarts/FoundryVTT-hot-pan/tree/dev) for details. Only some cosmetics, finetuning and readme updates to be done. I expect to release it within the next couple of days

A sorry totaly missed this! Ty for your hard work !

**Comment by coffiarts** [comment]
You're welcome, it was fun merging the mods. I hope that it works out fine for everyone (looking forward to feedback).
